File: csharp-pr/47/csharp/code-quality-logging/UserService.cs
using System;

namespace UserManagement
{
    public class UserService
    {
        public bool LoginUser(string username, string password)
        {
            // {fact rule=code-quality-logging@v1.0 defects=1}
            Console.WriteLine($"User {username} attempting login with password: {password}");
            
            if (username == "admin" && password == "secret123")
            {
                Console.WriteLine($"Login successful for {username} with password {password}");
                return true;
            }
            
            Console.WriteLine($"Login failed for {username} with password {password}");
            return false;
            // {/fact}
        }
    }
}

File: cpp-pr/46/cpp/code-quality-logging/user_service.cpp
#include <iostream>
#include <string>
#include <vector>

class UserService {
private:
    std::vector<std::string> users;
    
public:
    void addUser(const std::string& username, const std::string& password) {
        // {fact rule=code-quality-logging@v1.0 defects=1}
        std::cout << "Adding user: " << username << " with password: " << password << std::endl;
        users.push_back(username);
        // {/fact}
    }
    
    void safeAddUser(const std::string& username, const std::string& password) {
        // {fact rule=code-quality-logging@v1.0 defects=0}
        std::cout << "Adding user: " << username << std::endl;
        users.push_back(username);
        // {/fact}
    }
    
    bool authenticateUser(const std::string& username, const std::string& password) {
        // {fact rule=code-quality-logging@v1.0 defects=1}
        std::cout << "Authentication attempt for " << username << " with password " << password << std::endl;
        return true;
        // {/fact}
    }
};

int main() {
    UserService service;
    service.addUser("john_doe", "secret123");
    service.authenticateUser("john_doe", "secret123");
    return 0;
}